projects/mui-angular/button/button.component.ts

```diff
@@ -1,36 +1,82 @@
-import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
-
+import { Component, ElementRef, Input, ViewChild, HostListener, Renderer2 } from '@angular/core';
 
 @Component({
   selector: 'mui-button',
   template: `
-    <button class="mui-btn" #button>
+    <button class="mui-btn" #button [disabled]="disabled">
       <ng-content></ng-content>
+      <span class="mui-btn__ripple-container" #container><span class="mui-ripple" #ripple></span></span>
     </button>
   `
 })
 
-
-export class ButtonComponent implements OnInit {
+export class ButtonComponent {
   @ViewChild('button') button: ElementRef;
+  @ViewChild('ripple') ripple: ElementRef;
 
   @Input() variant: string = '';
   @Input() color: string = '';
   @Input() size: string = '';
```

Comment:
Should we make these inputs `optional?` as well?

Reply:
Yes.